app/controllers/answers_controller.rb

```diff
@@ -0,0 +1,71 @@
+class AnswersController < ApplicationController
+
+  def flag_favorite
+    @answer = Answer.find(params[:answer_id])
+    @question = Question.find(@answer.question_id)
+    # if the question is owned by the current_user update the value to true unless one already is true
+
+    if @question.has_favorite?
+      @question.answers.find_by(favorite: true).update_attributes(favorite: false)
+    end
+
+    if @question.user_id == current_user.id
+      @answer.update_attributes(favorite: true)
+    end
+
+    redirect_to :back
+  end
+
+  def new
+    @answer = Answer.new
+    @question = Question.find(params[:question_id])
+  end
+
+  def create
+    @answer = current_user.answers.new(answer_params.merge(question_id: params[:question_id]))
```

Comment:
I'd do
```
question = Question.find(params[:question_id]
answer = question.answers.build(answer_params)
```

Separately, have answer_params do
```
params.require(:answer).permit(:content, :question_id).merge(user: current_user)
```
so that the user_id is set on the new question